bundles/org.openhab.binding.nanoleaf/src/main/java/org/openhab/binding/nanoleaf/internal/OpenAPIUtils.java

```diff
@@ -162,11 +163,21 @@ public static boolean checkRequiredFirmware(@Nullable String modelId, @Nullable
         return true;
     }
 
-    private static int[] getFirmwareVersionNumbers(String firmwareVersion) throws IllegalArgumentException {
+    public static int[] getFirmwareVersionNumbers(String firmwareVersion) throws IllegalArgumentException {
+        LOGGER.debug("firmwareVersion: {}", firmwareVersion);
         Matcher m = FIRMWARE_VERSION_PATTERN.matcher(firmwareVersion);
         if (!m.matches()) {
-            throw new IllegalArgumentException("Malformed controller firmware version");
+            m = FIRMWARE_VERSION_PATTERN_BETA.matcher(firmwareVersion);
+            if (!m.matches()) {
+                throw new IllegalArgumentException("Malformed controller firmware version " + firmwareVersion);
+            }
```

Comment:
Just a suggestion to keep matching close to the actual pattern:
```suggestion
        Matcher m = FIRMWARE_VERSION_PATTERN.matcher(firmwareVersion);
        if (m.matches()) {
            return new int[] { Integer.parseInt(m.group(1)), Integer.parseInt(m.group(2)),
                    Integer.parseInt(m.group(3)) };
        } else {
            m = FIRMWARE_VERSION_PATTERN_BETA.matcher(firmwareVersion);
            if (m.matches()) {
                return new int[] { Integer.parseInt(m.group(1)), Integer.parseInt(m.group(2)), Integer.parseInt(m.group(3)),
                    Integer.parseInt(m.group(4)) };
            } else {
                throw new IllegalArgumentException("Malformed controller firmware version " + firmwareVersion);
            }
        }
```

Reply:
fine, I will change as proposed.